src/app/features/lp-form/reducer.ts

```diff
@@ -438,9 +446,15 @@ export const slice = createSlice({
         value: number | null;
       }>,
     ) => {
-      state.userInput.fixedUpper = value;
-      // whenever fixed upper changes -> update existing position
-      // updateExistingPosition(state);
+      const amm = state.amm;
+
+      let nextFixedRateUpperNumber: number | null = null;
+      if (amm !== null && value !== null) {
```

Comment:
why not stop function execution if no amm? for example on line 450 we can do if(!amm) { return; }